### PR TITLE
Dispatch 1585 - Add vhost aliases for configuration sharing

### PR DESCRIPTION
**Pull request by ChugR**
This PR defines:

- configuration attribute to specify the aliases
- policy code to enforce legal specifications
- configuration tests
- mock operational tests

This PR does not demonstrate actual AMQP Open performative hostnames getting policy to use aliased vhost policy settings. That is on the way.

### CODE REVIEW

File: python/qpid_dispatch/management/qdrouter.json [hunk 1]
```diff
@@ -1945,6 +1945,13 @@
                     "required": true,
                     "create": true
                 },
+                "aliases": {
+                    "type": "string",
+                    "description": "Alternate hostnames that share this vhost configuation. Hosts named in this attribute are treated as if this vhost was defined with the alias name in the vhost 'hostname' attribute. This attribute is implemented to help with multitennant configurations where multiple vhosts share a common configuration. The string is a comma- or space-separated list of literal hostnames or hostname patterns. A vhost aliases hostname must be unique across all vhost hostnames and all of their aliases.",
```
Comment (kgiusti) [inline python/qpid_dispatch/management/qdrouter.json:1950]: spelling "multiteNNant"

File: python/qpid_dispatch_internal/policy/policy_manager.py [hunk 2]
```diff
@@ -169,6 +180,23 @@ def set_max_message_size(self, size):
...
+
+#
+#
+#
+def policy_lookup_vhost_alias(mgr, vhost):
+    """
+    Look up a user in the policy database
+    Called by C code
+    @param mgr:
+    @param user:
```
Comment (kgiusti) [inline python/qpid_dispatch_internal/policy/policy_manager.py:192]: remove unused params?

File: docs/books/modules/user-guide/creating-vhost-policies.adoc [hunk 3]
```diff
@@ -54,6 +55,11 @@ The literal hostname of the vhost (the messaging endpoint) or a pattern that mat
 +
 If `enableVhostNamePatterns` is set to `true`, you can use wildcards to specify a pattern that matches a range of hostnames. For more information, see xref:vhost-policy-hostname-pattern-matching-rules-{context}[].
 
+`aliases`::
+Alternative literal hostnames or patterns that direct the router to use the settings in this vhost. Alias hostnames that match an incoming connection will use the settings defined in the contained vhost. In a multi-tenant configuration a connection to a vhost alias will use the base vhost hostname for the tenant namespace. In this example if a connection is directed to vhost `example.org` then the settings from the base vhost hostname `example.com` will apply and `example.com` will be the tenant namespace. Vhost `hostname` and `aliases` settings from all vhosts are shared in a single list and must be unique.
```
Comment (pwright) [inline docs/books/modules/user-guide/creating-vhost-policies.adoc:59]: Alternative literal hostnames or patterns that direct the router to use the settings in this vhost. 
Alias hostnames that match an incoming connection use the settings defined in the vhost section. 
In a multi-tenant configuration, a connection to a vhost alias uses the base vhost hostname for the tenant namespace. 
In this example if a connection is directed to vhost `example.org` then the settings from the base vhost hostname `example.com` apply and `example.com` becomes the tenant namespace. 
Vhost `hostname` and `aliases` settings from all vhosts must be unique.

@ChugR  Not sure if my second sentence is correct?

Comment (ChugR) [reply docs/books/modules/user-guide/creating-vhost-policies.adoc:59]: Yes, I think it's fine. Please submit that as a separate PR. Thanks!

**Pull request closed without merging**